### ISSUE TITLE
Uncaught Error: Offset 160 hasn't been loaded yet.

### ISSUE THREAD

**Issue by anzorb** [aadsm/jsmediatags#14]
Hi @aadsm, thanks for the awesome work :)

Found this issue, that only happens to a few files. It doesn't fire the error handler, but instead blows up, and I can't 'catch' it. Let me know how I can help.

```
new jsmediatags.Reader(filePath)
            .read({
                onSuccess: function(tag) {
                },
                onError: function(error) {
                }
            });
```

Here's the stack:
```
Uncaught Error: Offset 160 hasn't been loaded yet
.getByteAt @ ChunkedFileData.js?7fb3:176
getByteAt @ NodeFileReader.js?72bc:25
getBytesAt @ MediaFileReader.js?47b0:70
getStringWithCharsetAt @ MediaFileReader.js?47b0:154
readTextFrame @ ID3v2FrameReader.js?cb45:84
_readFrames @ ID3v2TagReader.js?b38e:147
_parseData @ ID3v2TagReader.js?b38e:70
self._loadData.onSuccess @ MediaTagReader.js?b084:43
```

For the sake of copyright, I've cropped the offending music file to ~15 seconds.

[Detroit Swindle - Figure of Speech_30secs.mp3.zip](https://github.com/aadsm/jsmediatags/files/181407/Detroit.Swindle.-.Figure.of.Speech_30secs.mp3.zip)

**Comment by aadsm** [comment]
Thanks, will look into it!

**Comment by wkande** [comment]
Love the package by the way.  I too am getting the same error on a file now and then, not all. The error handler is not called and my Node.js server blows up.  Here is my stack trace.  Also I have attached a file that errors. I assume its ID3 info is corrupted which causes the error.  If you could get the failure to manifest into the error handler that would be great.

[150406-153829 Sadhu Maharaja Jingle.mp3.zip](https://github.com/aadsm/jsmediatags/files/187926/150406-153829.Sadhu.Maharaja.Jingle.mp3.zip)



``****************** START uncaughtException ********************
Un-Caught exception (reported from server.js): 
[Error: Offset 167 hasn't been loaded yet.]
Error: Offset 167 hasn't been loaded yet.
    at ChunkedFileData.getByteAt (/Users/warren/Development/audio-streaming apis/node_modules/jsmediatags/build2/ChunkedFileData.js:176:11)
    at NodeFileReader.getByteAt (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/NodeFileReader.js:25:27)
    at NodeFileReader.getBytesAt (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/MediaFileReader.js:70:23)
    at NodeFileReader.getStringWithCharsetAt (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/MediaFileReader.js:154:22)
    at readTextFrame (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/ID3v2FrameReader.js:84:15)
    at ID3v2TagReader._readFrames (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/ID3v2TagReader.js:147:40)
    at ID3v2TagReader._parseData (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/ID3v2TagReader.js:70:25)
    at self._loadData.onSuccess (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/MediaTagReader.js:43:29)
    at processData (/Users/warren/Development/audio-streaming-apis/node_modules/jsmediatags/build2/NodeFileReader.js:83:7)
    at FSReqWrap.wrapper [as oncomplete] (fs.js:576:17)
****************** END uncaughtException ********************
``

**Comment by wkande** [comment]
An update to the problem I posted earlier.  For the file I posted the issue appears to be related to the "artist" tag When I exclude the artist tag everything works OK. I would guess the artist info embedded in the MP3 file is bad.  It the package could gracefully trap its own error and report such we would be there.

`new jsmediatags.Reader(file).setTagsToRead(["title", "year", "album", "year", "genre", "picture"])`

**Comment by simonmclean** [comment]
I'm getting this issue as well. Though it's not limited to particular offsets. I've run into the same problem with Offsets 50, 144, 153 162 and so on. Error being thrown from ChunkedFileData.js

**Comment by sosico** [comment]
I'm having the same problem with other offset methods. Uncaught Error: Offset 113595 hasn't been loaded yet. jsmediatags.min.js:9

**Comment by sosico** [comment]
Hi Antonio, I’m still having problems, can I send you the file that gives me the error?

jsmediatags.min.js:12 Uncaught Error: Offset 113608 hasn't been loaded yet.

**Comment by aadsm** [comment]
Yes please. You should be able to attach files to the comment.

On Tuesday, May 3, 2016, sosico <notifications@github.com> wrote:

> Hi Antonio, I’m still having problems, can I send you the file that gives
> me the error?
>
> jsmediatags.min.js:12 Uncaught Error: Offset 113608 hasn't been loaded yet.
>
>
>
>
> > On Apr 28, 2016, at 9:35 PM, António Afonso <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> >
> > Closed #14 <https://github.com/aadsm/jsmediatags/issues/14> via 2047eae
> <
> https://github.com/aadsm/jsmediatags/commit/2047eaed966a5aeef1c51f002e450b6cdf8f9d26
> >.
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly or view it on GitHub <
> https://github.com/aadsm/jsmediatags/issues/14#event-645097533>
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/aadsm/jsmediatags/issues/14#issuecomment-216588547>
>


-- 
Best regards,
António Afonso

**Comment by conceptualspace** [comment]
seeing this as well, in my case the track title contains a "+" character...

**Comment by conceptualspace** [comment]
[+Dennis Brown - Let Love In + dub.zip](https://github.com/aadsm/jsmediatags/files/344913/Dennis.Brown.-.Let.Love.In.dub.zip)

**Comment by wutzcrtoledo** [comment]
Hi, I've got the the issue. Love the library btw, much faster than others that I have tried, and also fixed an issue I had with old latin characters encoding I was struggling . 
In my opinion this is a concurrence issue. And a sync method would avoid the issue although it wouldn't work as fast as it works today. 
Why I say it's a concurrence issue it's because in my program I'm running an script that loop through "My Music" folder and gets ID3 info on each mp3 found. 
At first I looked that always crashed on the same mp3 file. So I looked the ID3 details on that file and also on the previous loaded mp3 and saw something interesting. The previous mp3 had a big image as part of its details. So, come up the theory that maybe  when the library was reading the id3 info of the mp3 that crashed, still was loading the heavy image of the previous mp3, causing a concurrence issue.
So part of the testing I did, was to first make a print of the id3 whole structure for each file. So this way the process takes a little bit longer before going to the next mp3 on the loop. 
The error continued, although now not in the same file. It started to crash in different places on the loop. 
So finally I've took out mp3 picture as part of the info only including the tags that I needed
 `new jsmediatags.Reader(mp3FilePath)
                  .setTagsToRead(["title", "album", "track"])
                  .read({
            onSuccess: function(tags) {
...
`
And from there worked fine.  So the theory is that if a tag is too big this one takes more than usual to load. So if meanwhile it's loading this file, an asynchronous call is made over the same object then you get this concurrence exception. 
I know is not a solution, just a theory, it's what I have and hope it helps someone.
